docs/paper/dev/api/event-api/chat-event.md

```diff
@@ -0,0 +1,152 @@
+
+```java title="ChatRenderer#render"
+public @NotNull Component render(@NotNull Player source, @NotNull Component sourceDisplayName, @NotNull Component message, @NotNull Audience viewer) {
+    // ...
+}
+```
+
+- The `render` method is called when a chat message is sent to the player.
+- The `source` parameter is the player that sent the message.
+- The `sourceDisplayName` parameter is the display name of the player that sent the message.
+- The `message` parameter is the message that was sent.
+- The `viewer` parameter is the player that is receiving the message.
+
+Now that we understand how the renderer works, we can start using it.
+
+## Using the Renderer
+
+There are two ways to use the renderer.
+1. Creating a new Class that implements `ChatRenderer`. 
+2. Using a lambda expression.
+
+Depending on the complexity of your renderer you may want to use one or the other.
+
+### Creating a new Class
+
+If you want to use a class for your renderer, you just need to implement the `ChatRenderer` interface.
+
+```java title="ExampleRenderer.java"
+public class ExampleRenderer implements ChatRenderer {
+
+    @Override
+    public @NotNull Component render(@NotNull Player source, @NotNull Component sourceDisplayName, @NotNull Component message, @NotNull Audience viewer) {
+        // ...
+    }
+
+}
+```
+
+To be able to use this renderer, we need to tell the event to use it.
+
+```java title="ChatListner.java"
+public class ChatListner implements Listener {
+
+    @EventHandler
+    public void onChat(AsyncChatEvent event) {
+        event.renderer(new ExampleRenderer());
+    }
+
+}
+```
+
+### Using a lambda expression
+
+Another way of using the renderer is by using a lambda expression.
+
+```java title="ChatListner.java"
+public class ChatListner implements Listener {
+
+    @EventHandler
+    public void onChat(AsyncChatEvent event) {
+        event.renderer((source, sourceDisplayName, message, viewer) -> {
+            // ...
+        });
+    }
+
+}
+```
+
+## Rendering the message
+
+Now that we have our renderer, we can start rendering the message.
+
+Let's say we want to render our chat to look like this:
+
+![](./assets/plain-message-rendering.png)
+
+To do this, we need to return a new `Component` that contains the message we want to send.
+
+```java title="ExampleRenderer.java"
+public class ExampleRenderer implements ChatRenderer {
+
+    @Override
+    public @NotNull Component render(@NotNull Player source, @NotNull Component sourceDisplayName, @NotNull Component message, @NotNull Audience viewer) {
+        return sourceDisplayName 
+                .append(Component.text(": ")) 
+                .append(message); 
+    }
+
+}
+```
+
+Now you can see that the message is rendered as we wanted it to be.
+
+## Rendering the message with a prefix using LuckPerms and MiniMessage
```

Comment:
don't really see why this is useful:
1) plugins already exist that do this including one shipped by LP
2) nobody would realistically be hardcoding the format like that, it should be using placeholders instead
3) now we have to maintain docs based on an entirely external plugin

would be better to make it generic, just having a "how to use a configurable format" and then just have the example be MiniMessage with placeholders and a little "add your own here!" bit or something